client/src/components/ShoeShowPage.js

```diff
@@ -5,88 +5,168 @@ import ReviewForm from "./ReviewForm.js"
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newReviewData)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
           const body = await response.json()
-          const updatedReviews = [...shoe.reviews, body.review]
-          setShoe({ ...shoe, reviews:updatedReviews})      
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
         }
+      } else {
+        const body = await response.json()
+        const updatedReviews = [...shoe.reviews, body.review]
+        setShoe({ ...shoe, reviews:updatedReviews})      
       }
-      catch(error) {
-        console.error(`Error in fetch: ${error.message}`)
+    }
+    catch(error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+  
+  useEffect(() => {
+      getShoe()
+  }, [])
+
+  let reviewFormComponent = ""
+  if (user){
+    reviewFormComponent = <ReviewForm postReview={postReview} shoe={shoe} errors={errors} setErrors={setErrors}/>
+  }
+
+  const deleteShoe = async () => {
+    try {
+      const response = await fetch(`/api/v1/shoes/${id}`, {
+        method: "DELETE",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(shoe)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      } else {
+        const body = await response.json()
+        setShouldRedirect(true)
       }
+    } catch (err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+  }
+
+  const handleDelete = () => {
+    if (confirm("Are you sure you want to delete this shoe? \nThis cannot be undone") == true) {
+      deleteShoe()
+    } else {
+      alert("This shoe has NOT been deleted.")
     }
-    
-    useEffect(() => {
-        getShoe()
-    }, [])
+  }
 
-    let reviewFormComponent = ""
-    if (user){
-      reviewFormComponent = <ReviewForm postReview={postReview} shoe={shoe} errors={errors} setErrors={setErrors}/>
+  if (shouldRedirect) {
+    return <Redirect push to="/" />
+  }
+
+  let deleteButton = ''
+  if (user?.is_admin == true) {
+    deleteButton = <button type="button" className="button" onClick={handleDelete}>Delete Shoe</button>
+  }
+
+  const deleteReview = async (reviewToDelete) => {
+    try {
+      const response = await fetch(`/api/v1/shoes/${id}/reviews/${reviewToDelete.id}`, {
```

Comment:
deleting a review does not need to know which shoe it belonged to, we just need the id of the review to delete - we should instead use the path
```js
fetch(`/api/v1/reviews/${reviewToDelete.id}`)
```